File: tests/CaseComment.cs
using System;

namespace Demo {
    public partial class CaseComment {
        public void Test(char val) {
            switch (val)
            {
                /* A */
                case 'A': Console.WriteLine("a"); break;
                /* B */
                case 'B': Console.WriteLine("b"); break;
            }
        }
    }
}

File: tests/CaseComment.pas
namespace Demo;

uses System;

type
  CaseComment = public class
  public
    method Test(val: Char);
    method CommentedBranch(val: Char);
  end;

implementation

method CaseComment.Test(val: Char);
begin
  case val of
    //A
    'A': Console.WriteLine('a');
    //B
    'B': Console.WriteLine('b');
  end;
end;

method CaseComment.CommentedBranch(val: Char);
begin
  case val of
    'A': Console.WriteLine('a');
//    'B':
//      Console.WriteLine('b');
    'C': Console.WriteLine('c');
  end;
end;

end.

